.github/workflows/create-github-release.yml

```diff
@@ -0,0 +1,39 @@
+#############################################################################
+# GitHub Action to create GitHub release.
+#
+#############################################################################
+name: "R: Create GitHub release"
+on:
+  workflow_dispatch:
+    inputs:
+      version:
+        description: 'Target Version without (-SNAPSHOT)'
+        required: true
+  workflow_call:
+    inputs:
+      version:
+        type: string
+        required: true
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.ref }}
+  cancel-in-progress: true
+
+permissions:
+  contents: write
+
+jobs:
+  publish:
+    name: Create GitHub release for ${{ inputs.version }}
```

Comment:
@stoyanK7 please point to where we are able to resolve name for jobs with context variables in the name; I do not see it here: https://github.com/stoyanK7/checkstyle/actions/runs/4193090241 . Most CI I have dealt with are not able to resolve variables in job/workflow names

Reply:
![image](https://user-images.githubusercontent.com/23459549/219378758-d9d38cf3-0f86-4810-be26-af369ad58f3a.png)

Is this what you mean? You can see the name on the top.

Reply:
Yes, I see that if job is successful we can resolve variable: https://github.com/stoyanK7/checkstyle/actions/runs/4177134639

Strange that we do not when it fails: https://github.com/stoyanK7/checkstyle/actions/runs/4193090241

Either way, if it works in successful run it is worth keeping.

Reply:
Example of failure to resolve:
![image](https://user-images.githubusercontent.com/31252532/219380713-4970b845-84b1-4611-8929-6fe0f351e4e4.png)